crates/wasmparser/src/validator.rs

```diff
@@ -1792,7 +1791,19 @@ mod arc {
                 return None;
             }
             debug_assert!(Arc::get_mut(&mut self.arc).is_some());
-            Some(unsafe { &mut *(&*self.arc as *const T as *mut T) })
+            Some(
+                // # Safety
+                //
+                // This operation is safe since the API asserts that
+                // `self.owned` is only ever `true` if there is just
+                // one owner of the `Arc`.
+                // As soon as `Arc::get_mut_unchecked` we should
+                // make use of it instead.
+                #[allow(clippy::cast_ref_to_mut)]
```

Comment:
I would prefer to have no `#[clippy]` attributes in this crate, can this be removed?

Reply:
How are we going to satisfy the `clippy` CI then?
Are those `#[clippy]` attributes okay in the crate root? Less ideal though ...

Reply:
We don't run clippy in CI in this repository, so there's nothing to satisfy.

Reply:
Is there a reason for that? With this PR the `wasmparser` crate could profit from having a `clippy` CI job at hand. In fact this PR adds a `clippy` CI job for `wasmparser`.

Reply:
Personally I do not believe that clippy adds value to a project. That's my personal opinion, others may feel differently. I don't really want to debate the merits of clippy here at this time, my point is that we don't run clippy and that is not something I would like to change off-hand in a PR.